### PR TITLE
Add more specific logic for ingress ACL and buffer profile

### PR DESCRIPTION
**Pull request by kcudnik**
Add more specific comparison logic for matching ingress ACL and matching buffer profile on queues and ingress priority groups

### CODE REVIEW

File: syncd/syncd_applyview.cpp [hunk 1]
```diff
@@ -3412,6 +3419,102 @@ std::shared_ptr<SaiObj> findCurrentBestMatchForBufferPool(
...
+        _In_ const std::shared_ptr<const SaiObj> &temporaryObj,
+        _In_ const std::vector<sai_object_compare_info_t> &candidateObjects)
+{
+    SWSS_LOG_ENTER();
+
+    /*
+     * For buffer profile we will try using SAI_INGRESS_PRIORITY_GROUP_ATTR_BUFFER_PROFILE
+     * or SAI_QUEUE_ATTR_BUFFER_PROFILE_ID for matching.
+     *
+     * We we are here, and buffer profile has assigned buffer pool, then buffer
```
Comment (lguohan) [inline syncd/syncd_applyview.cpp]: remove we

**Comment by lguohan** [comment]
do we need unit test for this comparison logic?

**Comment by lguohan** [comment]
code looks good to me, but I think we need to add unit tests for these two cases.

**Comment by kcudnik** [comment]
yes, i will add unittest with full log that Wenda provided which will also cover the empty Lag case, i will send  that in next pr

File: syncd/syncd_applyview.cpp (hunk 1 above)
Comment (kcudnik) [reply syncd/syncd_applyview.cpp]: fixed

**Comment by yxieca** [comment]
Made to 201811 branch on 2/14/2019

**Pull request merged**